### PR TITLE
adding principal type to RA Create Parameters

### PR DESCRIPTION
**Pull request by grlin**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by grlin** [comment]
@darshanhs90

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-09-01-preview/examples/DeleteRoleAssignmentById.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "parameters": {
+    "scope": "scope",
+    "roleId": "roleAssignmentId",
+    "api-version": "2018-01-01-preview"
```
Comment (darshanhs90) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-09-01-preview/examples/DeleteRoleAssignmentById.json]: could you change the api versions for all the examples

File: specification/authorization/resource-manager/readme.md [hunk 2]
```diff
@@ -115,6 +115,16 @@ input-file:
...
+These settings apply only when `--tag=package-2018-09-01-preview` is specified on the command line.
+
+``` yaml $(tag) == 'package-2018-09-01-preview'
+input-file:
+- Microsoft.Authorization/preview/2015-06-01/authorization-ClassicAdminCalls.json
+- Microsoft.Authorization/preview/2018-09-01-preview/authorization-RoleBasedCalls.json
+```
+
 ---
 # Code Generation
```
Comment (darshanhs90) [inline specification/authorization/resource-manager/readme.md]: add tag for GO like this
### Tag: package-2018-01-01-preview and go

These settings apply only when `--tag=package-2018-01-01-preview --go` is specified on he command line.
Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.

``` yaml $(tag) == 'package-2018-01-01-preview' && $(go)
output-folder: $(go-sdk-folder)/services/authorization/mgmt/2018-01-01-preview/authorization
```

File: specification/authorization/resource-manager/readme.md [hunk 3]
```diff
@@ -115,6 +115,16 @@ input-file:
...
+### Tag: package-2018-09-01-preview
+
+These settings apply only when `--tag=package-2018-09-01-preview` is specified on the command line.
+
+``` yaml $(tag) == 'package-2018-09-01-preview'
+input-file:
+- Microsoft.Authorization/preview/2015-06-01/authorization-ClassicAdminCalls.json
+- Microsoft.Authorization/preview/2018-09-01-preview/authorization-RoleBasedCalls.json
+```
+
```
Comment (darshanhs90) [inline specification/authorization/resource-manager/readme.md]: include your api version in GO multi api version
### Go multi-api

``` yaml $(go) && $(multiapi)
batch:
  - tag: package-2015-07
  - tag: package-2017-10-01-preview
  - tag: package-2018-01-01-preview
```

File: specification/authorization/resource-manager/readme.md (hunk 2 above)
Comment (grlin) [reply specification/authorization/resource-manager/readme.md]: Is there a reason this is only done for go and not Java etc?

**Comment by darshanhs90** [comment]
Looks good from my side

**Pull request closed without merging**